composer/trainer/trainer.py

```diff
@@ -47,6 +47,35 @@
 
 log = logging.getLogger(__name__)
 
+def _dynamic_microbatch_wrapper(func: Callable) -> Callable:
+    """Wraps function to catch CUDA Out of Memory Errors and adaptively change microbatch
```

Comment:
Is there a better place to put a wrapper? Maybe a util folder / fn? Making this part of the class is messy because the first argument then has to be self, which doesn't play nicely with decorators. I'm not sure of any workarounds where we can include this in the class, but maybe someone with more decorator knowledge knows how to do that

Reply:
It would be reasonable to just create a new file in `composer/trainer/` to hold dynamic grad accum related stuff. Even if it's just this one function.